### PR TITLE
Use npm scripts

### PR DESCRIPTION
**Pull request by SimenB**
Hey,

This is mostly making use of `npm`'s `script` feature.  It now lints before running tests, and you can uglify using `npm run uglify`. This should make the `Makefile` redundant, but I kept it in in case you prefer that. :smile: It also removes a dependency on a global `uglifyjs`.

I saw you also had `indent`-checking in JSHint. This isn't supported anymore (http://stackoverflow.com/a/23098280/1850276), so I added JSCS as a replacement for it.

### CODE REVIEW

File: .jscsrc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  //"validateIndentation": 2, index.js uses 2, and tests.js uses 4. https://github.com/jscs-dev/node-jscs/issues/1106
```
Comment (SimenB) [inline .jscsrc]: I don't know if you want to normalize your indentation. If you do, I'll be happy to give it a whack.

File: test/tests.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*global it: false, describe: false, DeepDiffConflict: false*/
+/*global DeepDiffConflict: false*/
```
Comment (SimenB) [inline test/tests.js]: Covered by `mocha` option in `.jshintrc`

File: test/tests.js [hunk 3]
```diff
@@ -360,35 +360,35 @@ describe('deep-diff', function() {
 
     describe('regression test for bug #10, ', function() {
         var lhs = {
-            "id": "Release",
-            "phases": [{
-                "id": "Phase1",
-                "tasks": [{
-                    "id": "Task1"
+            id: 'Release',
```
Comment (SimenB) [inline test/tests.js]: This si the only place you quote keys in objects, so I removed it, and added a JSCS rule for it.
Also `"` vs `'`

File: package.json [hunk 4]
```diff
@@ -15,6 +15,10 @@
     "sonstone",
     "ravishivt"
   ],
+  "files": [
+    "index.js",
+    "releases/"
+  ],
```
Comment (SimenB) [inline package.json]: This is what files are included when doing `npm install deep-diff`.

Comparison: http://imgur.com/X754Cwp,GIS3QPq

File: package.json [hunk 5]
```diff
@@ -28,9 +32,14 @@
...
+    "jscs": "^1.12.0",
+    "jshint": "^2.6.3",
+    "mocha": "^2.2.1",
+    "uglifyjs": "^2.4.10"
   },
   "scripts": {
-    "test": "mocha -R spec"
+    "uglify": "uglifyjs index.js -o releases/deep-diff-$npm_package_version.min.js  -r '$,require,exports,module,window,global' -m  --comments '/^!/'",
+    "pretest": "jscs index.js test/ && jshint index.js test/",
+    "test": "mocha"
```
Comment (SimenB) [inline package.json]: `spec` is default reporter

Comment (flitbit) [inline package.json]: Yep; I noticed that change a while ago - wasn't when this thing was born.

**Pull request merged**